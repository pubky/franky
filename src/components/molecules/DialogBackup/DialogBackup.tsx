'use client';

import * as Atoms from '@/atoms';
import * as Core from '@/core';
import Image from 'next/image';
import * as Molecules from '@/molecules';

export function DialogBackup() {
  const { mnemonic } = Core.useOnboardingStore();

  return (
    <Atoms.Dialog>
      <Atoms.DialogTrigger asChild>
        <Atoms.Button
          variant="outline"
          className="text-primary-foreground hover:text-primary-foreground text-xs font-bold border shadow-sm"
        >
          Backup
        </Atoms.Button>
      </Atoms.DialogTrigger>
      <Atoms.DialogContent className="sm:max-w-5xl gap-0 max-h-[90vh]">
        <Atoms.DialogHeader className="pr-6">
          <Atoms.DialogTitle>Back up your pubky</Atoms.DialogTitle>
        </Atoms.DialogHeader>
        <Atoms.Container className="h-full pr-4">
          <Atoms.Container className="gap-4">
            <Atoms.Typography size="sm" className="text-muted-foreground font-medium">
              Safely back up and store the secret seed for your pubky. Which backup method do you prefer? You can choose
              multiple backup methods if you wish.
            </Atoms.Typography>
            <Atoms.Container className="w-full flex md:flex-row flex-col gap-3">
              <Atoms.Card className="w-full p-6 bg-card rounded-lg flex flex-col gap-6">
                <Atoms.Typography size="md" className="text-base font-bold text-card-foreground leading-none">
                  Recovery phrase
                </Atoms.Typography>
                <Image src="/images/note.png" alt="Note" width={112} height={112} className="self-center" />
                <Molecules.DialogBackupPhrase />
              </Atoms.Card>
              <Atoms.Card className="w-full p-6 bg-card rounded-lg flex flex-col gap-6">
                <Atoms.Typography size="md" className="text-base font-bold text-card-foreground leading-none">
                  Download encrypted file
                </Atoms.Typography>
                <Image src="/images/folder.png" alt="Folder" width={112} height={112} className="self-center" />
                <Molecules.DialogBackupEncrypted />
              </Atoms.Card>
              <Atoms.Card className="w-full p-6 bg-card rounded-lg flex flex-col gap-6">
                <Atoms.Typography size="md" className="text-base font-bold text-card-foreground leading-none">
                  Export to Pubky Ring
                </Atoms.Typography>
                <Image src="/images/keyring.png" alt="Keys" width={112} height={112} className="self-center" />
                <Molecules.DialogExport mnemonic={mnemonic} />
              </Atoms.Card>
            </Atoms.Container>
          </Atoms.Container>
        </Atoms.Container>
      </Atoms.DialogContent>
    </Atoms.Dialog>
  );
}
