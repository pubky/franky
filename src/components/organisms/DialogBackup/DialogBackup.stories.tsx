import type { Meta, StoryObj } from '@storybook/nextjs-vite';
import { DialogBackup } from './DialogBackup';

const meta = {
  title: 'Molecules/DialogBackup',
  component: DialogBackup,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    open: { control: 'boolean' },
    onOpenChange: { action: 'openChange' },
  },
} satisfies Meta<typeof DialogBackup>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};

export const Controlled: Story = {
  args: {
    open: true,
    onOpenChange: (open: boolean) => console.log('Open changed:', open),
  },
};
