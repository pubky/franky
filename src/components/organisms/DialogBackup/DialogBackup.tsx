'use client';

import React from 'react';
import * as Atoms from '@/atoms';
import * as Core from '@/core';
import Image from 'next/image';
import * as Organisms from '@/organisms';

interface DialogBackupProps {
  open?: boolean;
  onOpenChange?: (open: boolean) => void;
}

interface BackupMethodCardProps {
  title: React.ReactNode;
  imageSrc: string;
  imageAlt: string;
  dialog: React.ReactNode;
}

function BackupMethodCard({ title, imageSrc, imageAlt, dialog }: BackupMethodCardProps) {
  return (
    <Atoms.Card className="w-full p-4 md:p-8 flex-col gap-4 md:gap-6">
      <Atoms.Typography size="md" className="text-base font-bold text-card-foreground leading-none">
        {title}
      </Atoms.Typography>
      <Image src={imageSrc} alt={imageAlt} width={112} height={112} className="self-center w-16 h-16 md:w-28 md:h-28" />
      {dialog}
    </Atoms.Card>
  );
}

export function DialogBackup({ open, onOpenChange }: DialogBackupProps = {}) {
  const { mnemonic } = Core.useOnboardingStore();

  return (
    <Atoms.Dialog open={open} onOpenChange={onOpenChange}>
      {open === undefined && (
        <Atoms.DialogTrigger asChild>
          <Atoms.Button
            id="backup-btn"
            variant="outline"
            className="text-primary-foreground hover:text-primary-foreground text-xs font-bold border shadow-sm"
          >
            Backup
          </Atoms.Button>
        </Atoms.DialogTrigger>
      )}
      <Atoms.DialogContent className="max-w-sm md:max-w-2xl" hiddenTitle="Back up your pubky">
        <Atoms.DialogHeader>
          <Atoms.DialogTitle id="backup-dialog-title">Back up your pubky</Atoms.DialogTitle>
          <Atoms.DialogDescription id="backup-dialog-description">
            Safely back up and store the secret seed for your pubky. Which backup method do you prefer? You can choose
            multiple backup methods if you wish.
          </Atoms.DialogDescription>
        </Atoms.DialogHeader>
        <Atoms.Container className="md:flex-row flex-col gap-3">
          <BackupMethodCard
            title={
              <>
                Recovery
                <br className="hidden md:inline" /> phrase
              </>
            }
            imageSrc="/images/note.png"
            imageAlt="Note"
            dialog={<Organisms.DialogBackupPhrase />}
          />
          <BackupMethodCard
            title="Download encrypted file"
            imageSrc="/images/folder.png"
            imageAlt="Folder"
            dialog={<Organisms.DialogBackupEncrypted />}
          />
          <BackupMethodCard
            title="Export to Pubky Ring"
            imageSrc="/images/keyring.png"
            imageAlt="Keys"
            dialog={<Organisms.DialogBackupExport mnemonic={mnemonic} />}
          />
        </Atoms.Container>
      </Atoms.DialogContent>
    </Atoms.Dialog>
  );
}
