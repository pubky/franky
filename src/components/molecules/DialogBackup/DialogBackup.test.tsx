import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { DialogBackup } from './DialogBackup';

// Mock atoms
vi.mock('@/atoms', () => ({
  Dialog: ({ children }: { children: React.ReactNode }) => <div data-testid="dialog">{children}</div>,
  DialogContent: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="dialog-content" className={className}>
      {children}
    </div>
  ),
  DialogHeader: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="dialog-header" className={className}>
      {children}
    </div>
  ),
  DialogTitle: ({ children }: { children: React.ReactNode }) => <h2 data-testid="dialog-title">{children}</h2>,
  DialogTrigger: ({ children, asChild }: { children: React.ReactNode; asChild?: boolean }) => (
    <div data-testid="dialog-trigger" data-as-child={asChild}>
      {children}
    </div>
  ),
  Container: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="container" className={className}>
      {children}
    </div>
  ),
  Typography: ({ children, size, className }: { children: React.ReactNode; size?: string; className?: string }) => (
    <p data-testid="typography" data-size={size} className={className}>
      {children}
    </p>
  ),
  Card: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="card" className={className}>
      {children}
    </div>
  ),
  Button: ({ children, variant, className }: { children: React.ReactNode; variant?: string; className?: string }) => (
    <button data-testid="button" data-variant={variant} className={className}>
      {children}
    </button>
  ),
}));

// Mock Next.js Image
vi.mock('next/image', () => ({
  default: ({
    src,
    alt,
    width,
    height,
    className,
  }: {
    src: string;
    alt: string;
    width: number;
    height: number;
    className?: string;
  }) => <img data-testid="image" src={src} alt={alt} width={width} height={height} className={className} />,
}));

describe('DialogBackup', () => {
  it('renders with correct structure', () => {
    render(<DialogBackup />);

    expect(screen.getByTestId('dialog')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-trigger')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-content')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-header')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-title')).toBeInTheDocument();
  });

  it('renders trigger button with correct text', () => {
    render(<DialogBackup />);

    const triggerButton = screen.getByText('Backup');
    expect(triggerButton).toBeInTheDocument();
    expect(triggerButton.tagName).toBe('BUTTON');
  });

  it('applies correct styling to trigger button', () => {
    render(<DialogBackup />);

    const triggerButton = screen.getByText('Backup');
    expect(triggerButton).toHaveClass('text-primary-foreground', 'hover:text-primary-foreground');
  });

  it('renders dialog title correctly', () => {
    render(<DialogBackup />);

    const title = screen.getByTestId('dialog-title');
    expect(title).toHaveTextContent('Back up your pubky');
    expect(title.tagName).toBe('H2');
  });

  it('renders description text', () => {
    render(<DialogBackup />);

    const description = screen.getByText(/Safely back up and store the secret seed/);
    expect(description).toBeInTheDocument();
  });

  it('renders three backup method cards', () => {
    render(<DialogBackup />);

    const cards = screen.getAllByTestId('card');
    expect(cards).toHaveLength(3);
  });

  it('renders recovery phrase card with correct content', () => {
    render(<DialogBackup />);

    expect(screen.getByText('Recovery phrase')).toBeInTheDocument();
    expect(screen.getAllByText('Continue')).toHaveLength(3);

    const noteImage = screen.getByAltText('Note');
    expect(noteImage).toHaveAttribute('src', '/images/note.png');
  });

  it('renders download encrypted file card with correct content', () => {
    render(<DialogBackup />);

    expect(screen.getByText('Download encrypted file')).toBeInTheDocument();

    const folderImage = screen.getByAltText('Folder');
    expect(folderImage).toHaveAttribute('src', '/images/folder.png');
  });

  it('renders export to Pubky Ring card with correct content', () => {
    render(<DialogBackup />);

    expect(screen.getByText('Export to Pubky Ring')).toBeInTheDocument();

    const keyringImage = screen.getByAltText('Keys');
    expect(keyringImage).toHaveAttribute('src', '/images/keyring.png');
  });

  it('applies correct styling to dialog content', () => {
    render(<DialogBackup />);

    const content = screen.getByTestId('dialog-content');
    expect(content).toHaveClass('sm:max-w-xl', 'gap-0');
  });

  it('applies correct styling to dialog header', () => {
    render(<DialogBackup />);

    const header = screen.getByTestId('dialog-header');
    expect(header).toHaveClass('pr-6');
  });

  it('applies correct styling to cards', () => {
    render(<DialogBackup />);

    const cards = screen.getAllByTestId('card');
    cards.forEach((card) => {
      expect(card).toHaveClass('w-full', 'p-6', 'bg-card', 'rounded-lg', 'flex', 'flex-col', 'gap-6');
    });
  });

  it('applies correct styling to card titles', () => {
    render(<DialogBackup />);

    const cardTitles = screen.getAllByText(/Recovery phrase|Download encrypted file|Export to Pubky Ring/);
    cardTitles.forEach((title) => {
      expect(title).toHaveClass('text-base', 'font-bold', 'text-card-foreground', 'leading-none');
    });
  });

  it('applies correct styling to images', () => {
    render(<DialogBackup />);

    const images = screen.getAllByTestId('image');
    images.forEach((image) => {
      expect(image).toHaveClass('w-28', 'h-28');
    });
  });

  it('renders all required elements', () => {
    render(<DialogBackup />);

    expect(screen.getByTestId('dialog')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-trigger')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-content')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-header')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-title')).toBeInTheDocument();
    expect(screen.getAllByTestId('card')).toHaveLength(3);
    expect(screen.getAllByTestId('image')).toHaveLength(3);
    expect(screen.getAllByText('Continue')).toHaveLength(3);
  });
});
