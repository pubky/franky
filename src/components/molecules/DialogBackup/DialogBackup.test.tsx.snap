// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DialogBackup > matches snapshot for card titles 1`] = `
<div
  data-testid="dialog"
>
  <div
    data-as-child="true"
    data-testid="dialog-trigger"
  >
    <button
      class="text-primary-foreground hover:text-primary-foreground"
      data-testid="button"
      data-variant="outline"
    >
      Backup
    </button>
  </div>
  <div
    class="sm:max-w-xl gap-0"
    data-testid="dialog-content"
  >
    <div
      class="pr-6"
      data-testid="dialog-header"
    >
      <h2
        data-testid="dialog-title"
      >
        Back up your pubky
      </h2>
    </div>
    <div
      class="h-full pr-4 overflow-y-auto"
      data-testid="container"
    >
      <div
        class="gap-4"
        data-testid="container"
      >
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          Safely back up and store the secret seed for your pubky. Which backup method do you prefer? You can choose multiple backup methods if you wish.
        </p>
        <div
          class="w-full flex md:flex-row flex-col gap-3"
          data-testid="container"
        >
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Recovery phrase
            </p>
            <div
              class="self-center"
              data-alt="Note"
              data-height="112"
              data-src="/images/note.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Note
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Download encrypted file
            </p>
            <div
              class="self-center"
              data-alt="Folder"
              data-height="112"
              data-src="/images/folder.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Folder
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Export to Pubky Ring
            </p>
            <div
              class="self-center"
              data-alt="Keys"
              data-height="112"
              data-src="/images/keyring.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Keys
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DialogBackup > matches snapshot for cards 1`] = `
<div
  data-testid="dialog"
>
  <div
    data-as-child="true"
    data-testid="dialog-trigger"
  >
    <button
      class="text-primary-foreground hover:text-primary-foreground"
      data-testid="button"
      data-variant="outline"
    >
      Backup
    </button>
  </div>
  <div
    class="sm:max-w-xl gap-0"
    data-testid="dialog-content"
  >
    <div
      class="pr-6"
      data-testid="dialog-header"
    >
      <h2
        data-testid="dialog-title"
      >
        Back up your pubky
      </h2>
    </div>
    <div
      class="h-full pr-4 overflow-y-auto"
      data-testid="container"
    >
      <div
        class="gap-4"
        data-testid="container"
      >
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          Safely back up and store the secret seed for your pubky. Which backup method do you prefer? You can choose multiple backup methods if you wish.
        </p>
        <div
          class="w-full flex md:flex-row flex-col gap-3"
          data-testid="container"
        >
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Recovery phrase
            </p>
            <div
              class="self-center"
              data-alt="Note"
              data-height="112"
              data-src="/images/note.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Note
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Download encrypted file
            </p>
            <div
              class="self-center"
              data-alt="Folder"
              data-height="112"
              data-src="/images/folder.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Folder
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Export to Pubky Ring
            </p>
            <div
              class="self-center"
              data-alt="Keys"
              data-height="112"
              data-src="/images/keyring.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Keys
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DialogBackup > matches snapshot for dialog content 1`] = `
<div
  data-testid="dialog"
>
  <div
    data-as-child="true"
    data-testid="dialog-trigger"
  >
    <button
      class="text-primary-foreground hover:text-primary-foreground"
      data-testid="button"
      data-variant="outline"
    >
      Backup
    </button>
  </div>
  <div
    class="sm:max-w-xl gap-0"
    data-testid="dialog-content"
  >
    <div
      class="pr-6"
      data-testid="dialog-header"
    >
      <h2
        data-testid="dialog-title"
      >
        Back up your pubky
      </h2>
    </div>
    <div
      class="h-full pr-4 overflow-y-auto"
      data-testid="container"
    >
      <div
        class="gap-4"
        data-testid="container"
      >
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          Safely back up and store the secret seed for your pubky. Which backup method do you prefer? You can choose multiple backup methods if you wish.
        </p>
        <div
          class="w-full flex md:flex-row flex-col gap-3"
          data-testid="container"
        >
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Recovery phrase
            </p>
            <div
              class="self-center"
              data-alt="Note"
              data-height="112"
              data-src="/images/note.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Note
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Download encrypted file
            </p>
            <div
              class="self-center"
              data-alt="Folder"
              data-height="112"
              data-src="/images/folder.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Folder
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Export to Pubky Ring
            </p>
            <div
              class="self-center"
              data-alt="Keys"
              data-height="112"
              data-src="/images/keyring.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Keys
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DialogBackup > matches snapshot for dialog header 1`] = `
<div
  data-testid="dialog"
>
  <div
    data-as-child="true"
    data-testid="dialog-trigger"
  >
    <button
      class="text-primary-foreground hover:text-primary-foreground"
      data-testid="button"
      data-variant="outline"
    >
      Backup
    </button>
  </div>
  <div
    class="sm:max-w-xl gap-0"
    data-testid="dialog-content"
  >
    <div
      class="pr-6"
      data-testid="dialog-header"
    >
      <h2
        data-testid="dialog-title"
      >
        Back up your pubky
      </h2>
    </div>
    <div
      class="h-full pr-4 overflow-y-auto"
      data-testid="container"
    >
      <div
        class="gap-4"
        data-testid="container"
      >
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          Safely back up and store the secret seed for your pubky. Which backup method do you prefer? You can choose multiple backup methods if you wish.
        </p>
        <div
          class="w-full flex md:flex-row flex-col gap-3"
          data-testid="container"
        >
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Recovery phrase
            </p>
            <div
              class="self-center"
              data-alt="Note"
              data-height="112"
              data-src="/images/note.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Note
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Download encrypted file
            </p>
            <div
              class="self-center"
              data-alt="Folder"
              data-height="112"
              data-src="/images/folder.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Folder
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Export to Pubky Ring
            </p>
            <div
              class="self-center"
              data-alt="Keys"
              data-height="112"
              data-src="/images/keyring.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Keys
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DialogBackup > matches snapshot for images 1`] = `
<div
  data-testid="dialog"
>
  <div
    data-as-child="true"
    data-testid="dialog-trigger"
  >
    <button
      class="text-primary-foreground hover:text-primary-foreground"
      data-testid="button"
      data-variant="outline"
    >
      Backup
    </button>
  </div>
  <div
    class="sm:max-w-xl gap-0"
    data-testid="dialog-content"
  >
    <div
      class="pr-6"
      data-testid="dialog-header"
    >
      <h2
        data-testid="dialog-title"
      >
        Back up your pubky
      </h2>
    </div>
    <div
      class="h-full pr-4 overflow-y-auto"
      data-testid="container"
    >
      <div
        class="gap-4"
        data-testid="container"
      >
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          Safely back up and store the secret seed for your pubky. Which backup method do you prefer? You can choose multiple backup methods if you wish.
        </p>
        <div
          class="w-full flex md:flex-row flex-col gap-3"
          data-testid="container"
        >
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Recovery phrase
            </p>
            <div
              class="self-center"
              data-alt="Note"
              data-height="112"
              data-src="/images/note.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Note
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Download encrypted file
            </p>
            <div
              class="self-center"
              data-alt="Folder"
              data-height="112"
              data-src="/images/folder.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Folder
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Export to Pubky Ring
            </p>
            <div
              class="self-center"
              data-alt="Keys"
              data-height="112"
              data-src="/images/keyring.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Keys
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DialogBackup > matches snapshot for trigger button 1`] = `
<div
  data-testid="dialog"
>
  <div
    data-as-child="true"
    data-testid="dialog-trigger"
  >
    <button
      class="text-primary-foreground hover:text-primary-foreground"
      data-testid="button"
      data-variant="outline"
    >
      Backup
    </button>
  </div>
  <div
    class="sm:max-w-xl gap-0"
    data-testid="dialog-content"
  >
    <div
      class="pr-6"
      data-testid="dialog-header"
    >
      <h2
        data-testid="dialog-title"
      >
        Back up your pubky
      </h2>
    </div>
    <div
      class="h-full pr-4 overflow-y-auto"
      data-testid="container"
    >
      <div
        class="gap-4"
        data-testid="container"
      >
        <p
          class="text-muted-foreground font-medium"
          data-size="sm"
          data-testid="typography"
        >
          Safely back up and store the secret seed for your pubky. Which backup method do you prefer? You can choose multiple backup methods if you wish.
        </p>
        <div
          class="w-full flex md:flex-row flex-col gap-3"
          data-testid="container"
        >
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Recovery phrase
            </p>
            <div
              class="self-center"
              data-alt="Note"
              data-height="112"
              data-src="/images/note.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Note
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Download encrypted file
            </p>
            <div
              class="self-center"
              data-alt="Folder"
              data-height="112"
              data-src="/images/folder.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Folder
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
          <div
            class="w-full p-6 bg-card rounded-lg flex flex-col gap-6"
            data-testid="card"
          >
            <p
              class="text-base font-bold text-card-foreground leading-none"
              data-size="md"
              data-testid="typography"
            >
              Export to Pubky Ring
            </p>
            <div
              class="self-center"
              data-alt="Keys"
              data-height="112"
              data-src="/images/keyring.png"
              data-testid="image"
              data-width="112"
            >
              Next.js Image: 
              Keys
            </div>
            <button
              data-testid="button"
            >
              Continue
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
